### ISSUE TITLE
Port IOApp.Simple to CE2

### ISSUE THREAD

**Issue by kubukoz** [typelevel/cats-effect#1721]
In series/3.x, a simpler interface for IOApp is provided, for the users who don't need args or exit codes:

```scala
object Main extends IOApp.Simple:
  def run: IO[Unit] = IO.println("hello")
```

I think it could be backported to CE2 for the poor souls who won't be able to migrate for a while for reasons.

**Comment by djspiewak** [comment]
Great idea!

**Comment by alexandrustana** [comment]
I could  try and solve this one

**Comment by djspiewak** [comment]
Go for it!

**Comment by kubukoz** [comment]
Closed in #1759, no idea why github didn't figure that one out.
